### PR TITLE
[WIP] [RFC] Add support for registering custom log levels

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds some WIP code to get the discussion going on adding a new public API method for registering a custom log level with structlog.

## Background, Context

There have been many discussions and questions on the internet on how to register a custom log level with method (it's a fairly common request and problem). There are also open issues on that in this repo - e.g. #47.

I've used monkey patching / workaround similar to the ones proposed in that ticket elsewhere myself for many years now, but I think it would be good to expose this functionality directly in struclog and as such, make it less hacky, more official and less fragile.

## Proposed Implementation

This implementation is not finished yet.

It's just initial basic proposal to get the discussions going. Once we agree on the approach, I'm happy to fully implement it (with tests and other things of course).

### TODO

- [ ] Agree on the approach
- [ ] Implement it
- [ ] Add support for custom log levels to ConsoleRender

### CODE REVIEW

**Comment by hynek** [comment]
This belongs to `structlog.stdlib` but otherwise: is it code you use in prod like this?

I mean it's very gross 😅 but it is so due to `logging`'s design…I'd want to think over the addition of the logging methods but that's secondary.

I guess we should ask for feedback on #47?

**Comment by hynek** [comment]
One idea before I forget it:

I think it would be better, to instead of globally monkeypatching stdlib.BoundLogger, to return a new stdlib.BoundLogger class.

File: src/structlog/_base.py [hunk 1]
```diff
@@ -205,3 +205,50 @@ def _proxy_to_logger(self, method_name, event=None, **event_kw):
...
+        def log_method(self, msg, *args, **kwargs):
+            return self.log(level_value, msg, *args, **kwargs)
+        log_method.__name__ = name
+        return log_method
+
+    log_method = make_logger_function(level_name_lower)
+
+    setattr(structlog.stdlib._FixedFindCallerLogger, level_name_lower, log_method)
+    setattr(structlog.stdlib.BoundLogger, level_name_lower, log_method)
+
```
Comment (danilito19) [inline src/structlog/_base.py:252]: I used some of this code for my project and I had to also add this line to get this to work
`setattr(logging.Logger, level_name_lower, log_method)`

**Comment by hynek** [comment]
I think this is stale. I'm interested in a solution of this problem tho.

**Pull request closed without merging**